scipy/sparse/csgraph/_shortest_path.pyx

```diff
@@ -1546,9 +1537,14 @@ cdef void insert_node(FibonacciHeap* heap,
     #              - node is a valid pointer
     #              - node is not the child or sibling of another node
     if heap.min_node:
-        add_sibling(heap.min_node, node)
         if node.val < heap.min_node.val:
+            # heap.min_node is always at the leftmost end of the linked-list
+            node.left_sibling = NULL
+            node.right_sibling = heap.min_node
+            heap.min_node.left_sibling = node
```

Comment:
```suggestion
            # Replace node with heap.min_node, which is always 
            # at the leftmost end of the roots' linked-list.
            node.left_sibling = NULL
            node.right_sibling = heap.min_node
            heap.min_node.left_sibling = node
```

Reply:
Yes, but did you mean "Replace heap.min_node with node"?

Reply:
Ah yes, my bad. I just have corrected it.